Readme.md

```diff
@@ -0,0 +1,181 @@
+2. Azure Bastion provides SSH connectivity to the jumpbox virtual machine.
+3. The administrator on the jumpbox tries to access the storage account via the Azure CLI. The jumpbox queries DNS for the public Azure Blob Storage Account endpoint: storageaccountname.blob.core.windows.net. Private DNS ultimately resolves to storageaccountname.privatelink.blob.core.windows.net, returning the private IP address of the private link endpoint which is 10.0.2.5 in this example.
+4. A private connection to the storage account is established through the private link endpoint.
+
+## Getting Started
+
+There are a couple of prerequisites you will need:
+
+- A service principal. You will need the service principal app id, object id and password
+- A SSH key pair
+
+### Creating a service principal
+
+You must pass the object id of a service principal as a parameter. This SP will be given Storage Blob Data Contributor Role Assignment at the Resource Group level. This SP can be used to connect to the storage accounts.
+
+To create a service principal, follow the guidance on [creating a password-based authenticated sp here](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli#password-based-authentication). The following is an example:
+
+```bash
+az ad sp create-for-rbac --scopes /subscriptions/mySubscriptionID \
+  --name myServicePrincipalName \
+  --role rolename
+
+az ad sp create-for-rbac --scopes "/subscriptions/mySubscriptionID" --role "Contributor" --name mySpName
+{
+  "appId": "appIdGuid",
+  "displayName": "mySpName",
+  "name": "guid",
+  "password": "password",
+  "tenant": "myTenantId"
+}
+```
+
+You will also need the object id of the SP. You can get that with this Azure CLI call
+
+```bash
+az ad sp show --id <appIdGuid from above> --query id
+```
+
+### Creating SSH key pair
+
+There are a variety of ways to create the SSH pey pair:
+
+- [Generate and store SSH keys in the Azure portal](https://docs.microsoft.com/azure/virtual-machines/ssh-keys-portal)
+- [Create and manage SSH keys locally](https://docs.microsoft.com/azure/virtual-machines/linux/create-ssh-keys-detailed)
+- [Create and manage SSH Keys with the Azure CLI](https://docs.microsoft.com/azure/virtual-machines/ssh-keys-azure-cli)
+
+## Deploying
+
+### Deploying Azure Front Door Standard
+
+Use the following command to deploy the Azure Front Door Standard deployment
+
+```bash
+az deployment sub create --template-file deployRgFrontDoorAndWeb.bicep \
+  --location centralus \
+  -p frontDoorSkuName=Standard_AzureFrontDoor assetPrefix=uniqueprefixname
+```
+
+### Deploying Azure Front Door Premium With Private Endpoints
+
+Use the following command to deploy the Azure Front Door Premium deployment
+
+```bash
+az deployment sub create --template-file deployRgFrontDoorAndWeb.bicep \
+  --location centralus \
+  -p frontDoorSkuName=Premium_AzureFrontDoor assetPrefix=uniqueprefixname
+```

```

Comment:
So, why two paths?  Doesn't the premium with private endpoints already show the complete meta pattern and the other is simply a subset of the meta pattern?  Seems like you could simply consolidate to the premium pattern since it's a superset.  Thoughts?

Reply:
Resolved in our convo!